runtime/lua/vim/lsp/handlers.lua

```diff
@@ -101,6 +101,30 @@ M['window/showMessageRequest'] = function(_, _, params)
   end
 end
 
+--@see https://microsoft.github.io/language-server-protocol/specifications/specification-current/#client_registerCapability
+M['client/registerCapability'] = function(_, _, params, client_id)
+  local client = vim.lsp.get_client_by_id(client_id)
+  local registrations = params['registrations']
+  for _, registration in pairs(registrations) do
+    local method = registration.method
+    local server_capability = vim.lsp._request_name_to_server_capability[method]
+    if server_capability then
+      client.resolved_server_capabilities[server_capability] = true;
```

Comment:
Does this always succeed ? is it possible as per the spec that server_capability is not in client.resolved_server_capabilities ? Maybe log the registration just in case.

Reply:
If this was the case, then client.resolved_server_capabilities[server_capability] would just add a new entry to the table, but since vim.lsp._request_name_to_server_capability[method] always maps to a key already supported in the table, this should always work.

There's another, bigger problem that I just realized. 

`client.resolved_server_capabilities[server_capability]` is not always a boolean, it can also take the form of a table.

See this example from the spec.

```json
{
	"method": "client/registerCapability",
	"params": {
		"registrations": [
			{
				"id": "79eee87c-c409-4664-8102-e03263673f6f",
				"method": "textDocument/willSaveWaitUntil",
				"registerOptions": {
					"documentSelector": [
						{ "language": "javascript" }
					]
				}
			}
		]
	}
}
```

In this particular example, this PR has the correct behavior as textDocument/willSaveWaitUnti is not in  `vim.lsp._request_name_to_server_capability`, so the conditional fails. However, type_definition can be of type `boolean | TypeDefinitionOptions | TypeDefinitionRegistrationOptions`, so this PR would overwrite whatever value with true, and ignore registerOptions which is undesirable.

I need to double check that it's always ok to just replace whatever is in client.resolved_server_capabilities[server_capability] with the table returned by register options